AWSCore/Authentication/AWSSignature.h

```diff
@@ -16,6 +16,7 @@
 #import <Foundation/Foundation.h>
 #import "AWSNetworking.h"
 
+NS_ASSUME_NONNULL_BEGIN
```

Comment:
Seems like there are three functions in AWSSignatureSignerUtility that can return null.. specifically, I think we should have:
```
+ (NSData * _Nullable)hash:(NSData *)dataToHash;
+ (NSString * _Nullable)hexEncode:(NSString *)string;
+ (NSString * _Nullable)HMACSign:(NSData *)data withKey:(NSString *)key usingAlgorithm:(uint32_t)algorithm;
```

Also, it seems like we can remove the following function prototype because there doesn't seem to be a corresponding implementation:
```
-+ (NSString *)getCanonicalizedQueryStringForDate:(NSDate *)currentDate;
```

Reply:
@jmwoo2 Since you appear to be well-versed in this code, and have investigated what changes _should_ be included, can we expect an official PR from Amazon soon? We forked here to get around those warnings, but would much rather use the official repo.

Reply:
Thanks for the vote of confidence, but I'm currently in the beginning stages of getting up to speed on this codebase.  I happen to be digging through github tickets and this particular one caught my eye, and happen to dive deep in this specific class.  As stated by palpatim@ [on this thread](https://github.com/aws-amplify/aws-sdk-ios/issues/2001), we need to audit nullability through the entire SDK, and there's currently no ETA on a fix.